src/components/AllocationsComponent.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { Header, Table, Pagination, Container } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import loadAllocationsAction from '../_actions/allocations.actions';
+import SideMenuComponent from '../_components/SideMenuComponent';
+import TableRowComponent from './TableRowComponent';
+import LoaderComponent from './LoaderComponent';
+import AllocationActionComponent from './AllocationActionComponent';
+import '../_css/AllocationsComponent.css';
+
+export class AllocationsComponent extends Component {
+  state = {
+    currentAllocations: [],
+    offset: 0,
+    activePage: 1,
+    limit: 10
+  }
+
+  componentDidMount() {
+    this.props.loadAllocationsAction();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.allAllocations.length !== prevProps.allAllocations.length) {
+      this.setTableContent();
+    }
+  }
+
+  getTotalPages = () => Math.ceil(this.props.allAllocations.length / this.state.limit);
+
+  setTableContent = () => {
+    const currentAllocations = this.props.allAllocations.slice(this.state.offset,
+      (this.state.activePage * this.state.limit));
+    // format date
+    const dateOptions = {
+      weekday: 'long',
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric'
+    };
+    currentAllocations.map(el => el.formatted_date = new Date(el.created_at).toLocaleDateString('en-US', dateOptions));
+    this.setState({
+      currentAllocations
+    });
+  }
+
+  calculateOffset = (activePage, limit) => (activePage - 1) * limit;
+
+  handlePaginationChange = (event, { activePage }) => {
+    this.setState({
+      activePage,
+      offset: this.calculateOffset(activePage, this.state.limit)
+    }, () => this.setTableContent());
+  }
+
+  render() {
+    if (this.props.isLoading) {
+      return (
+        <SideMenuComponent>
+          <LoaderComponent size="large" dimmerStyle={{ height: '100vh' }} />
+        </SideMenuComponent>
+      );
+    } else if (!this.props.isLoading && _.isEmpty(this.props.allAllocations)) {
```

Comment:
Use of `else if`  is unnecessary due to the return statements. Having it as
```
if (!this.props.isLoading && _.isEmpty(this.props.allAllocations)) {
  ...
}
```
will suffice